### DISCUSSION TITLE
Rolling up currency values with more than 2 decimal places in a multi-currency org

### DISCUSSION THREAD

**Discussion by marcusjsk** [SFDO-Community/declarative-lookup-rollup-summaries#1101] [Q&A]
Hi all,

Running into a strange issue when trying to roll up a value into a custom currency field in a multi-currency environment.

The standard currency field on the child record has a value of 0.035
When I roll-up this value on the parent record is pulls a value of 0.032

The currency for these records is not the master currency for the org
The org multi-currency decimal is 3 for the field with the target currency (not the standard 2)
The target field is currency with 3 decimal places

The roll-up config 
![Screen Shot 2021-10-29 at 11 13 44](https://user-images.githubusercontent.com/42288876/139409620-71ccfefd-b43b-402f-b729-07ce824d5a78.png)

Has anyone seen or run into this issue?  Any other ideas on why this could be happening?

**Comment by aheber** [comment]
@marcusjsk, thanks for reporting this issue. I'll do my best to provide an answer but we do our best support from the Trailhead Group, https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p so in the future you may want to start there.

Without looking at your org I'd say you're experiencing the same thing as #859. The problem in this case is often if the record currency is not the native currency of the org. We use aggregate queries (the database sums the value for us) to retrieve the rollup amounts. A Salesforce behavior in those scenarios is that it will convert foreign currencies into the native currency for the org before summing the value. This often mean that currency conversion occurs and loses granularity as part of the process, this causes slight discrepancies in currency rollups from time to time. If you think this applies to you then I'd suggest weighing in on this issue, #284. If I remember correctly that is the change that when completed will help to correct problems like this.

If you don't think scenario applies to you please provide some additional detail and I can see if we can figure it out. Likely the problem is occurring at that database aggregation level, maybe just for different reasons than stated above.

You can test some of this yourself by running a SOQL query, something like `SELECT PARENTID__c, SUM(Currency_Field__c) FROM CHILDOBJECT__c GROUP BY PARENTID__c`, swapping out the PARENTID__C, Currency_Field__c, and CHILDOBJECT__c placeholders for something legitimate to your situation.

Let me know what you find and we can go from there.

**Comment by marcusjsk** [comment]
Thanks for the feedback!

I went ahead and used a number field to work around the issue.


Cheers,
Marcus
